M2/Macaulay2/d/ctype.d

```diff
@@ -52,10 +51,29 @@ export isspace    (c:char):bool := (chartype(c) & SPACE    ) != 0;
 export isnewline  (c:char):bool := (chartype(c) & NEWLINE  ) != 0;
 export isquote    (c:char):bool := (chartype(c) & QUOTE    ) != 0;
 
-export isvalidsymbol  (s:string):bool := (
-     if int(uchar(s.0)) == 226 && length(s) == 3 then return true; -- ugly unicode math symbol hack
+-- c = two bytes concatenated
+export ismathoperator(c:int):bool := (
+    (c & 0xffe0) == 0xc2a0 || -- latin-1 punctuation/symbols
+    c == 0xc397            || -- multiplication sign
+    c == 0xc3b7            || -- division sign
+    c == 0xd188            || -- sha (shuffle product)
```

Comment:
Unfortunately 0xd188 (ш) is a ... Russian letter. you can't declare unilaterally that one letter in some alphabet is a mathematical symbol.
In any case at this stage it wouldn't do the user any good, since they can't define binary operators.
Once they can, they can name their opertaors ш, or привет, or any other sequence of characters they like.

Reply:
actually my comment is not entirely correct. without the line I singled out, they still won't be able to write `1ш2`. but I think that's just the way it is, as I said we can't prevent Russian users from naming their variables `шутник`...

Reply:
I went ahead and removed ш for now.  We can cross that bridge later if we really want it as a keyword